packages/node_modules/nav-frontend-skjema/src/bekreft-checkboks-panel.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import * as PT from 'prop-types';
+import { guid } from "nav-frontend-js-utils";
+
+export interface Props {
+    onChange: (event: React.SyntheticEvent<EventTarget>) => void;
+    checked: boolean;
+    label: string;
+    children?: React.ReactNode | React.ReactChildren;
+    inputProps?: React.InputHTMLAttributes<HTMLInputElement>;
+    className?: string;
+}
+
+export interface State {
+    hasFocus: boolean;
+    isHover: boolean;
+}
+
+export class BekreftCheckboksPanel extends React.Component<Props, State> {
+    private childrenId: string;
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {hasFocus: false, isHover: false};
+        if (this.props.children) {
+            this.childrenId = guid();
+        }
+    }
+
+    toggleOutline() {
+        this.setState({
+            ...this.state,
+            hasFocus: !this.state.hasFocus,
+        });
+    }
+
+    toggleHover(isHover: boolean) {
```

Comment:
Trenger ikke et parameter for å styre denne tenker jeg, sett heller bare til `!this.state.isHover`. :)

Reply:
Basert på bruken bør den nok heller renames imho, e.g 
```
setHover(isHover: boolean) {
  return () => this.setState({ isHover });
}
```

Reply:
Har oppdatert koden slik som Nicklas foreslår. Grunnen til at jeg ikke benyttet !this.state.isHover var at jeg ikke var helt sikker på hva dom skjer dersom musen ligger over komponenten når siden lastes. Er ikke sikkert det er noe problem.

Reply:
Er litt i tvil om hele panelet bør få en hover-stil på den måten når det bare er checkboksen og labelen som er klikkbar. Antar du har løst det slik pga. [designet](https://zpl.io/aXGgR0l) som ligger i Zeplin, men lurer litt på om det designet ble laget med en antagelse om at hele panelet skulle være klikkbar. Er det slik dere har løst det også @erlendev?

Skal høre med Sergio hva han mener om dette. Ellers knallbra PR :)

Reply:
Ble løst slik med tanke på designet slik det er i Zeplin  ja @Lillebo. Om det bare er checkboks og label som skal ha hover så slipper man også å bruke javascript til å sette hover. Si fra når du har snakket med Sergio, så endrer vi på det. Kan også tenkes at han mener at innholdet i øvre del skal være alignet med starten på checkboksen :)

Reply:
Snakket med Sergio og vi ble enige om å begrense hover-stilen til bare checkboksen og label. La samtidig til en liten transition på panelet for å fade fra orange til grønn. Måtte gjøre endringene for å vise ham hvordan det ville bli, så committer like greit disse. Fint om du tar en titt @chfinst :)

Han vil også gjerne at innholdet i øvre del alignes med starten på checkboksen ja. Det har jeg ikke fikset.

Reply:
Flott! Har aligna innholdet i øvre del med starten på checkboksen nå. Løste det ved å legge på -2rem i margin, hva tror du om det @Lillebo ?

La også til litt padding rundt label-teksten for å få litt større klikkflate.

Reply:
Råbra 👍